cf-custom-resources/lib/nlb-cert-validator-updater.js

```diff
@@ -400,6 +412,135 @@ async function activateOption(option, loadBalancerDNS, loadBalancerHostedZone) {
     }).promise();
 }
 
+/**
+ * Retrieve all certificates used for the service and cache the results.
+ * @returns {Array<Object>} An array of descriptions for the certificates used by the service.
+ */
+async function serviceCertificates() {
+    if (cachedServiceCertificates) {
+        return cachedServiceCertificates;
+    }
```

Comment:
I wonder if it's worthwhile introducing caching state (probably of bugs due to caching vs. performance). I think given that we expect 1 certificate usually for a service, then I think it's safe to remove caching. Let me know if there is a particular reason why you opted into this approach!

Reply:
To delete the certificate, we need to give the certificateARN, which can be obtained from `resourceGroupTaggingAPI`, or `ListCertificate`. I cache this to avoid an extra API call to `resourceGroupTaggingAPI` 🤔

Reply:
IMO this is unlikely but my worry with caching is that it's possible someone creates a certificate manually with the tags and then on function invocation we have the wrong list. I feel like it's okay to write dumb code here as latency isn't of real concern.
I'd rest better if only clients are global across invocations, but not stateful data.

Reply:
Gotcha gotcha. That makes sense to me!

To to clarify:
>  it's possible someone creates a certificate manually with the tags and then on function invocation we have the wrong list

This is to say the list would be wrong it the certificate is created manually in between API calls right? Like after it's cached, and before the next API call.